core/src/main/java/io/onetable/delta/DeltaSchemaExtractor.java

```diff
@@ -181,9 +181,7 @@ private OneSchema toOneSchema(
       case "timestamp":
         type = OneType.TIMESTAMP;
         // Timestamps in Delta are microsecond precision by default
-        metadata =
-            Collections.singletonMap(
-                OneSchema.MetadataKey.TIMESTAMP_PRECISION, OneSchema.MetadataValue.MICROS);
+        metadata.put(OneSchema.MetadataKey.TIMESTAMP_PRECISION, OneSchema.MetadataValue.MICROS);
```

Comment:
singletonMap is very efficient in its implementation or am I missing something?

Reply:
Yes, what you say is reasonable, thank you for your review